### ISSUE TITLE
feat: Tabs as a child page

### ISSUE THREAD

**Issue by jpmckearin** [driftyco/ionic#5334]
**Type**: <span ionic-type>feat</span>

**Ionic Version**: <span ionic-version>2.x</span>

**Platform**: <span ionic-platform>mobile</span>  <span ionic-webview>webview</span>

<span ionic-description>
Use case:
Master-detail pages where the master is a list of items and the details view is broken into different tabs. Currently, a tabs page has it's own nav stack for each individual tab using the [root] property. This would allow navigation into and out of a tabs detail page back to a master page. This was discussed during development of Ionic 1 but was decided to be put off for Ionic 2 due to limitations with ui-router.
</span>

<span is-issue-template></span>

**Comment by adamdbradley** [comment]
Have you tried this with Ionic2? This should be able to work or I may be reading it wrong. An ion-tabs is no different than an ion-page as far as ion-nav is concerned, and you push and pop different ion-tabs onto the stack.

**Comment by jpmckearin** [comment]
@adamdbradley is there an Ionic 2 codepen starter somewhere that you know of? I'll throw one together if not

**Comment by jgw96** [comment]
This should be possible if im understanding whats being asked correctly. @jpmckearin would a codepen starter be easier for you? I can throw that together if needed.

**Comment by jpmckearin** [comment]
@jgw96 I think a couple of codepen starters, like we have for ionic v1, would make it much easier to communicate issues in the future. I'm just strapped for time at the moment

**Comment by lordgreg** [comment]
This is exactly the same issue as I've already said:

- https://github.com/driftyco/ionic/issues/5196 and
- https://github.com/driftyco/ionic/issues/5172

But, @adamdbradley didn't understood what I wanted... :-1:

**Comment by jgw96** [comment]
@lordgreg @jpmckearin you are able to simply push() a tabs "page" in Ionic 2 with the new navigation. Check out my repo GitNinja for an example of how to go from a simple page to a tabs page in Ionic 2.

**Comment by lordgreg** [comment]
@jgw96 I've just checked your repo and haven't found any tabs implemented at all. However, you're missing the point here; There's no problem pushing Tab page to navigation stack. It is bugged because of current Angular2 Routing service implementation.

This is the exact same issue I've posted [here](https://github.com/driftyco/ionic/issues/5196). Check the project I've made inside the link, and you will see where the problem is.
